src/Microsoft.DotNet.XHarness.Common/Utilities/FileUtils.cs

```diff
@@ -3,30 +3,42 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using System.IO;
-using System.Threading.Tasks;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.DotNet.XHarness.Common.Utilities;
 
 public static class FileUtils
 {
-    public static string FindFileInPath(string engineBinary)
+    private static readonly string[] s_extensions = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                                                        ? new[] { ".exe", ".cmd", ".bat" }
+                                                        : new[] { "" };
+
+    public static (string fullPath, string? errorMessage) FindExecutableInPATH(string filename)
```

Comment:
Probably just a matter of preference, but wouldn't it be more natural to throw an exception, like `FileNotFoundException`, instead of returning a tuple?

Reply:
I wanted the control of what to do if the file is not found to be with the caller.

Reply:
I changed the code to follow the Try* pattern.

Reply:
Yeah, I understood. What I meant was throwing and catching it later to produce correct exit code, but the Try* pattern looks good 👍